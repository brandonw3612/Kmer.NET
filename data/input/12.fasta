>beginning of sequence: ACG (10)
ACGACGACGACGACGACGACGACGACGACG
TAGCTCGATGCATCGATACG
>end of sequence: CGA (10)
ACGTCGATGCATCGATACGT
CGACGACGACGACGACGACGACGACGACGA
>ssr followed by an ssr CG (8) AT (8)
TGCATCGA
CGCGCGCGCGCGCGCG
ATATATATATATATAT
GCATGACT
>ssr within an ssr ACGTCTC (5) w/ TC (2) inside each
TGCATCG
ACGTCTC
ACGTCTC
ACGTCTC
ACGTCTC
ACGTCTC
GCATGACT
>ssrs overlapping within last period (first one is bigger period size)
TGCATCGT
GAC
GAC
GAC
GAC
GAC
GAC
ACACACACACACACACAC
GCATGACT
>ssrs overlapping entire last period (second one is bigger period size)
TGCATCGT
GAC
GAC
GAC
GAC
GAC
GAC
GACTGACTGACTGACTGACTGACTGACTGACT
GCATGACT
>ssrs overlapping more than last period (second one is bigger period size)
TGCATCT
GAC
GAC
GAC
GAC
GAC
GAC
CGAC
CGAC
CGAC
CGAC
CGAC
CGAC
CGAC
CGAC
CGAC
GCATGACT
>ssrs overlapping within first period (second one is bigger period size)
TGCATCT
GAGAGAGAGAGAGAGA
GAC
GAC
GAC
GAC
GAC
GAC
GCATGACT
>ssrs overlapping entire first period (first one is bigger period size)
TGCATCGA
TGACTGACTGACTGACTGACT
GAC
GAC
GAC
GAC
GAC
GAC
GAC
GAC
GCATGACT
>ssrs overlapping more than first period (first one is bigger period size)
TGCATCGA
TGACG
TGACG
T
GAC
GAC
GAC
GAC
GAC
GCATGACT
>imperfect ssr (ATCGATCT should not register as ATCG x2)
ATCG
ATCT
>atomicity check: AT (6) not ATAT (3) or ATATAT (2)
TGCATCGC
ATATATATATAT
GCATGACT
>identical ssrs separated by 1 nucleotide
TGCATCT
AGAGAGAGAGAGAGAGAGAG
T
AGAGAGAGAGAGAGAGAGAG
TCATGACT
>identical ssrs separated by 2 nucleotides
TGCATCT
AGAGAGAGAGAGAGAGAGAG
TC
AGAGAGAGAGAGAGAGAGAG
TCATGACT
>identical ssrs separated by 3 nucleotides
TGCATCT
AGAGAGAGAGAGAGAGAGAG
TGC
AGAGAGAGAGAGAGAGAGAG
TCATGACT
>identical ssrs separated by 4 nucleotides
TGCATCT
AGAGAGAGAGAGAGAGAGAG
TGCT
AGAGAGAGAGAGAGAGAGAG
TCATGACT
>identical ssrs separated by 5 nucleotides
TGCATCT
AGAGAGAGAGAGAGAGAGAG
TGCAT
AGAGAGAGAGAGAGAGAGAG
TCATGACT
>identical ssrs separated by 6 nucleotides
TGCATCT
AGAGAGAGAGAGAGAGAGAG
TGCATC
AGAGAGAGAGAGAGAGAGAG
TCATGACT
>identical ssrs separated by 7 nucleotides
TGCATCT
AGAGAGAGAGAGAGAGAGAG
TGCATGT
AGAGAGAGAGAGAGAGAGAG
TCATGACT
>report first of ATATATAT vs TATATATA in seq. ...ATATATATA...
TGCATCG
ATATATATATA
GCATGACT
>SSR1SSR2SSSR3SSR2SSR1
ACGTACGTACGTACGTACGTACGTACGTACGT
GGGGGGGGGGGGGGGGGGG
CGTACGTACGTACGTACGTACGTACGTACGTA
CCCCCCCCCCCCCCCCCCC
ACGTACGTACGTACGTACGTACGTACGTACGT
>...SSR1SSR2SSSR3SSR2SSR1...
TGCATCGC
ACGTACGTACGTACGTACGTACGTACGTACGT
GGGGGGGGGGGGGGGGGGG
CGTACGTACGTACGTACGTACGTACGTACGTA
CCCCCCCCCCCCCCCCCCC
ACGTACGTACGTACGTACGTACGTACGTACGT
GCATGACT
>SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1
ACGTACGTACGTACGTACGTACGTACGTACGT
GGGGGGGGGGGGGGGGGGG
CGTACGTACGTACGTACGTACGTACGTACGTA
CCCCCCCCCCCCCCCCCCC
ACGTACGTACGTACGTACGTACGTACGTACGT
GCAGTAG
ACGTACGTACGTACGTACGTACGTACGTACGT
GGGGGGGGGGGGGGGGGGG
CGTACGTACGTACGTACGTACGTACGTACGTA
CCCCCCCCCCCCCCCCCCC
ACGTACGTACGTACGTACGTACGTACGTACGT
>...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...
TGCATCGC
ACGTACGTACGTACGTACGTACGTACGTACGT
GGGGGGGGGGGGGGGGGGG
CGTACGTACGTACGTACGTACGTACGTACGTA
CCCCCCCCCCCCCCCCCCC
ACGTACGTACGTACGTACGTACGTACGTACGT
GCAGTAG
ACGTACGTACGTACGTACGTACGTACGTACGT
GGGGGGGGGGGGGGGGGGG
CGTACGTACGTACGTACGTACGTACGTACGTA
CCCCCCCCCCCCCCCCCCC
ACGTACGTACGTACGTACGTACGTACGTACGT
GCATGACT
>backtrack
ATATATATATATATAT
TATAT
TATAT
>backtrack2
ATATATATATATATAT
TATATAT
TATATAT
