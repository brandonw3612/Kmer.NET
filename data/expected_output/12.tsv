#Sequence_Name	SSR	Repeats	Position
beginning of sequence: ACG (10)	ACG	10	0
end of sequence: CGA (10)	CGA	10	20
ssr followed by an ssr CG (8) AT (8)	CG	8	8
ssr followed by an ssr CG (8) AT (8)	AT	8	24
ssr within an ssr ACGTCTC (5) w/ TC (2) inside each	ACGTCTC	5	7
ssrs overlapping within last period (first one is bigger period size)	GAC	6	8
ssrs overlapping within last period (first one is bigger period size)	AC	10	24
ssrs overlapping entire last period (second one is bigger period size)	GACT	8	26
ssrs overlapping entire last period (second one is bigger period size)	GAC	7	8
ssrs overlapping more than last period (second one is bigger period size)	CGAC	10	21
ssrs overlapping more than last period (second one is bigger period size)	GAC	6	7
ssrs overlapping within first period (second one is bigger period size)	GAC	6	23
ssrs overlapping within first period (second one is bigger period size)	GA	9	7
ssrs overlapping entire first period (first one is bigger period size)	TGAC	6	8
ssrs overlapping entire first period (first one is bigger period size)	GAC	8	29
ssrs overlapping more than first period (first one is bigger period size)	TGACG	3	8
ssrs overlapping more than first period (first one is bigger period size)	GAC	5	19
atomicity check: AT (6) not ATAT (3) or ATATAT (2)	AT	6	8
identical ssrs separated by 1 nucleotide	AG	10	7
identical ssrs separated by 1 nucleotide	AG	10	28
identical ssrs separated by 2 nucleotides	AG	10	7
identical ssrs separated by 2 nucleotides	AG	10	29
identical ssrs separated by 3 nucleotides	AG	10	7
identical ssrs separated by 3 nucleotides	AG	10	30
identical ssrs separated by 4 nucleotides	AG	10	7
identical ssrs separated by 4 nucleotides	AG	10	31
identical ssrs separated by 5 nucleotides	AG	10	7
identical ssrs separated by 5 nucleotides	AG	10	32
identical ssrs separated by 6 nucleotides	AG	10	7
identical ssrs separated by 6 nucleotides	AG	10	33
identical ssrs separated by 7 nucleotides	AG	10	7
identical ssrs separated by 7 nucleotides	AG	10	34
report first of ATATATAT vs TATATATA in seq. ...ATATATATA...	AT	5	7
SSR1SSR2SSSR3SSR2SSR1	ACGT	8	0
SSR1SSR2SSSR3SSR2SSR1	CGTA	8	51
SSR1SSR2SSSR3SSR2SSR1	ACGT	8	102
SSR1SSR2SSSR3SSR2SSR1	G	19	32
SSR1SSR2SSSR3SSR2SSR1	C	19	83
...SSR1SSR2SSSR3SSR2SSR1...	ACGT	8	8
...SSR1SSR2SSSR3SSR2SSR1...	CGTA	8	59
...SSR1SSR2SSSR3SSR2SSR1...	ACGT	8	110
...SSR1SSR2SSSR3SSR2SSR1...	G	19	40
...SSR1SSR2SSSR3SSR2SSR1...	C	19	91
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	ACGT	8	0
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	CGTA	8	51
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	ACGT	8	102
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	ACGT	8	141
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	CGTA	8	192
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	ACGT	8	243
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	G	19	32
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	C	19	83
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	G	19	173
SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1	C	19	224
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	ACGT	8	8
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	CGTA	8	59
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	ACGT	8	110
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	ACGT	8	149
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	CGTA	8	200
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	ACGT	8	251
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	G	19	40
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	C	19	91
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	G	19	181
...SSR1SSR2SSSR3SSR2SSR1...SSR1SSR2SSSR3SSR2SSR1...	C	19	232
backtrack	AT	8	0
backtrack	TATAT	3	11
backtrack2	AT	8	0
backtrack2	TATATAT	3	9
